### ISSUE TITLE
AutoFactory omits Provider in constructor call 

### ISSUE THREAD

**Issue by mkaulig** [google/auto#296]
The following code generates a factory which does not compile due to a wrong parameter in the constructor call.

```java
import javax.inject.Provider;
import com.google.auto.factory.AutoFactory;
import com.google.auto.factory.Provided;

@AutoFactory
public class Root {
    public Root(Provider<Child> child, @Provided Injected injected) {
    }
}
```

The create method in the generated factory expects a parameter of type `Child` instead of `Provider<Child>` and calls the constructor with the wrong parameter.

```java
import javax.annotation.Generated;
import javax.inject.Inject;
import javax.inject.Provider;
@Generated("com.google.auto.factory.processor.AutoFactoryProcessor")
public final class RootFactory {
  private final Provider<Injected> injectedProvider;
  @Inject
  public RootFactory(Provider<Injected> injectedProvider) {
    this.injectedProvider = injectedProvider;
  }
  public Root create(Child child) {
    return new Root(child, injectedProvider.get());
  }
}
```

This bug seems to be only triggered when using an instance of javax.inject.Provider.

**Comment by foxylion** [comment]
Any updates on this? I'm having the same issues.

**Comment by foxylion** [comment]
If someone needs a workaround. It is possible to use `com.google.inject.Provider` instead of `javax.inject.Provider`. Auto Factory will then generate a valid factory.

**Comment by ronshapiro** [comment]
I'm submitting a change for this internally now. Thanks for alerting this of the bug!

**Comment by foxylion** [comment]
@eamonnmcmanus Thanks for the fix. When is this planned to be release on maven central?

**Comment by ronshapiro** [comment]
I just released `1.0-beta4` which has this: https://oss.sonatype.org/content/repositories/public/com/google/auto/factory/auto-factory/
